### PR TITLE
[Hub Generated] Review request for Microsoft.Security to add version stable/2020-01-01

### PR DESCRIPTION
**Pull request by nitsi**
If you are a MSFT employee you can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/nitsi/azure-rest-api-specs/dev-security-Microsoft.Security-2020-01-01-alerts...master/).

### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by ArcturusZhang** [comment]
Hi @nitsi is this PR ready for review? If so, given that you have the permission to edit labels, it would be best to add a `WaitForARMReview` label.

**Comment by ArcturusZhang** [comment]
You have inconsistency enum definitions between [here](https://github.com/Azure/azure-rest-api-specs/blob/f01882efa05a57fb3b461ac5df3feaa1faca6e02/specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/assessmentMetadata.json#L361) and the `severity` in the new `alerts.json` file, which is causing all SDK generation to fail.

File: specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/alerts.json
Comment (ArcturusZhang) [inline specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/alerts.json]: The `severity` has already been defined [here](https://github.com/Azure/azure-rest-api-specs/blob/f01882efa05a57fb3b461ac5df3feaa1faca6e02/specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/assessmentMetadata.json#L361), and it is inconsistency with the definition here (with different values). Maybe changing another word for this enum will work, or updating the existing swagger as well to make these two definitions align with each other.

**Review by ArcturusZhang — CHANGES_REQUESTED** [review]

File: specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/alerts.json
Comment (nitsi) [reply specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/alerts.json]: Your second here is not a link, can you fix this?

Comment (ArcturusZhang) [reply specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/alerts.json]: The second `here` just means the code of this comment. Or if you insist, [here](https://github.com/nitsi/azure-rest-api-specs/blob/f01882efa05a57fb3b461ac5df3feaa1faca6e02/specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/alerts.json#L653)

Comment (nitsi) [reply specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/alerts.json:653]: Ok, got it.
I can't change it to match the assessment API:

1.  I want to keep as much backwards compatibility as passable.

Comment (nitsi) [reply specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/alerts.json:653]: Ok, got it.
I can't change it to match the assessment API:
1) I want to keep as much backwards compatibility as passable with alerts 2019-01-01 [here](https://github.com/Azure/azure-rest-api-specs/blob/f01882efa05a57fb3b461ac5df3feaa1faca6e02/specification/security/resource-manager/Microsoft.Security/stable/2019-01-01/alerts.json#L575). The new version is to align our schema with our partners, today we export our alerts in this schema to ASI (Sentinel), log analytics, workspace automation and some other export features. So I wish to agile the severity to them. 
2) We have alerts with "information" severity while assessment don't

Comment (nitsi) [reply specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/alerts.json:669]: I see now that the python code generation can't have 2 different definitions with the same name. I'll check with the assessment team to see if we can find a solution

**Comment by ArcturusZhang** [comment]
There is another enum called `Status` is blocking the SDK generation, would you please take a look and provide a fix for this? @nitsi

File: specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/alerts.json
Comment (nitsi) [reply specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/alerts.json]: Fixed the issue

File: specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/alerts.json
Comment (anthony-c-martin) [inline specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/alerts.json:956]: Please model resource identifiers using the standard ARM id string format rather than writing a custom model for it. This has the following benefits:
1. Consistency with other ARM services.
2. Allows other services (ARM, ARG, ARM Deployments) to programatically understand resource dependencies.
3. Avoids users of the SDK to have to write resourceId parsing (e.g. if a user wants to fetch a loganalytics resource and pass the id to this field).

File: specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/alerts.json
Comment (anthony-c-martin) [inline specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/alerts.json:575]: Why is `additionalProperties` being added? Do you expect properties outside of what is being modeled in this spec?

File: specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/alerts.json
Comment (anthony-c-martin) [inline specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/alerts.json:690]: [nit] "Possible Values: Informational, Low, Medium, and High." is unnecessary here in the description as you're modeling the possible values in the enum.

File: specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/alerts.json
Comment (anthony-c-martin) [inline specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/alerts.json:860]: There are other fields named `startTimeUtc` & `endTimeUtc`. Please could you name all your date-time fields consistently (either with the `Utc` suffix, or without it)?

File: specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/alerts.json
Comment (anthony-c-martin) [inline specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/alerts.json:845]: [nit] "Supported statuses are: Active, Resolved, Dismissed." is unnecessary as these values are already described by the enum.

File: specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/alerts.json
Comment (anthony-c-martin) [inline specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/alerts.json:592]: Why `"additionalProperties": true` here?

File: specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/alerts.json
Comment (anthony-c-martin) [inline specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/alerts.json:604]: Why `"additionalProperties": true` here?

File: specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/alerts.json
Comment (anthony-c-martin) [inline specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/alerts.json:609]: Why `"additionalProperties": true` here?

**Review by anthony-c-martin — CHANGES_REQUESTED** [review]

**Comment by ArcturusZhang** [comment]
Hi @nitsi could you please have a look and resolve the ARM's comments?

**Comment by nitsi** [comment]
Sorry guys there was a priority shift, I'll be back on this PR by Monday EOD

**Comment by ArcturusZhang** [comment]
Hi @nitsi any updates?

File: specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/alerts.json
Comment (yibirnba) [reply specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/alerts.json:690]: Removed from comment

File: specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/alerts.json
Comment (yibirnba) [reply specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/alerts.json:845]: Removed from comment

File: specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/alerts.json
Comment (yibirnba) [reply specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/alerts.json]: Removed

File: specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/alerts.json
Comment (yibirnba) [reply specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/alerts.json]: Entities can have more inner properties, this is still changing and not constant. Change to additional with type string

File: specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/alerts.json
Comment (yibirnba) [reply specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/alerts.json]: The user can add which property he like to this property bag. Change to additional with type string

**Comment by ArcturusZhang** [comment]
All CI passed, should be good to merge once ARM signed off

**Review by ArcturusZhang — APPROVED** [review]

**Comment by yibirnba** [comment]
@shblum

File: specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/alerts.json
Comment (yibirnba) [reply specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/alerts.json:865]: Done

File: specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/alerts.json
Comment (yibirnba) [reply specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/alerts.json]: Changed to type string

File: specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/alerts.json
Comment (yibirnba) [reply specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/alerts.json:961]: Hi, can you explain " standard ARM id string format" ? 
ResourceIdentifiers has 2 formatting types, which property "type" is common , and each has its own additional properties. 
Do you mean to include only the type property as string, and allow additional properties?

Comment (majastrz) [reply specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/alerts.json:961]: I think Anthony is talking about using the standard ARM resource ID string (something like `/subscriptions/{id}/resourceGroups/{rg}/providers...`) You are basically doing that with Azure Resources, but I definitely see the challenge with LA which is partially in the data plane (due to agent IDs).

---
In reply to: [437166909](https://github.com/Azure/azure-rest-api-specs/pull/8805#discussion_r437166909) [](ancestors = 437166909)

File: specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/alerts.json
Comment (majastrz) [inline specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/alerts.json:805]: >The resource identifiers for this alert [](start = 26, length = 39)

This doesn't seem accurate. The resource identifier for the alert is in the `id` property of the alert resource. I think you mean something else here. Would `impactedResources`, `affectedResources` or something like that be a better fit?

File: specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/alerts.json
Comment (majastrz) [reply specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/alerts.json:961]: We definitely should drop "Identifier" from the definition name and the property above (see comment above). It can be confused with the meaning of the `id` property which is the resource identifier.

---
In reply to: [439029078](https://github.com/Azure/azure-rest-api-specs/pull/8805#discussion_r439029078) [](ancestors = 439029078,437166909)

File: specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/alerts.json
Comment (yibirnba) [reply specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/alerts.json:805]: It is an array of all the identifiers for the resource: we can have AADIdentifier (which has AadTenantId inside) , AzureResource ( which has AzureResourceId inside) , LogAnalytics ( which has WorkspaceId, WorkspaceSubscriptionId and some more identifiers inside).
So the name - resourceIdentfiers will hold all the identifiers for this resource. not just the "resourceId" itself

File: specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/alerts.json
Comment (yibirnba) [reply specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/alerts.json:961]: See my comment on your other comment about the resourceIdentifiers. It is not just the "ARMResourceId" , there are some identifiers for the resource , ARMResourceId is identifier in terms of AzureResource. There are some more resource identifiers for the resource

File: specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/alerts.json
Comment (majastrz) [reply specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/alerts.json:805]: Yes, I got that. All I'm saying is that the description text of "The resource identifiers for this alert" is confusing. The text is ambiguous and implies that the alert is being identified by the `resourceIdentifiers` property. I think part of the issue is the name of the property (schema of the property is fine) I think the name of the name should be changed to something that better captures the relationship between the alert and the resources. Maybe `impactedResources`, `affectedResources`, or something like that?

---
In reply to: [439062511](https://github.com/Azure/azure-rest-api-specs/pull/8805#discussion_r439062511) [](ancestors = 439062511)

File: specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/alerts.json
Comment (majastrz) [reply specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/alerts.json:961]: Replied on that one. I'm disagreeing with Anthony and agreeing with you, but have a concern about naming.

---
In reply to: [439064304](https://github.com/Azure/azure-rest-api-specs/pull/8805#discussion_r439064304) [](ancestors = 439064304)

File: specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/alerts.json
Comment (yibirnba) [reply specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/alerts.json:805]: Changed the description, and removed "for this alert". Do you thing it is more clear now?. We use this name internally in schema which we expose to out partners so changing it might be confusing.

**Comment by majastrz** [comment]
@chiragg4u can you take over ARM review on this one also?

File: specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/examples/Alerts/GetAlertResourceGroupLocation_example.json
Comment (chiragg4u) [inline specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/examples/Alerts/GetAlertResourceGroupLocation_example.json:31]: Why did we modeled this differently? Can't we use azureResourceId as workspace is also an Azure resource?

Please work with Gaurav Bhatnagar for continuing the review.

Comment (yibirnba) [reply specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/examples/Alerts/GetAlertResourceGroupLocation_example.json:31]: Those are different values , or did you mean something else? I'll work with Gaurav

File: specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/alerts.json
Comment (yibirnba) [reply specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/alerts.json:639]: alertType is not a closed list, there are hundreds of types, and new types are been added often

File: specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/examples/Alerts/GetAlertResourceGroupLocation_example.json
Comment (yibirnba) [reply specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/examples/Alerts/GetAlertResourceGroupLocation_example.json:31]: we keep this separately on different properties:  WorkspaceId, WorkspaceSubscriptionId, WorkspaceResourceGroup, AgentId , since those are not mandatory fields and we want to show information that we have

**Comment by ravbhatnagar** [comment]
Signing off from ARM side.

**Comment by ravbhatnagar** [comment]
Adding new APIs requires a new api-version as per the new versioning guidelines. If you need or think this must be added to the existing api-version, please get sign off from Azure API review board. Adrian hall, Johan Stenberg etc. can help

**Comment by ravbhatnagar** [comment]
Removing ARM Review label as APIs are fine from ARM standpoint.

**Comment by ArcturusZhang** [comment]
Hi @nitsi is this PR ready to merge? This LGTM and should be good to merge. If this is ready to merge, please remove the `DoNotMerge` label and let me know. Thanks

**Comment by ArcturusZhang** [comment]
Hi @nitsi do we have any update on this? Please let me know if this is ready to merge

**Comment by nitsi** [comment]
> Hi @nitsi do we have any update on this? Please let me know if this is ready to merge

Not yet, we still have a few things to do before:
1) Tests 
2) Manifest deployment/update 

We will give you an update once we are ready

**Comment by nitsi** [comment]
@ArcturusZhang Please marge

**Review by ArcturusZhang — APPROVED** [review]

**Pull request merged**
